pkg/component/types.go

```diff
@@ -28,7 +28,7 @@ type Component struct {
 	Handle
 	Name     string
 	CR       metav1.Object
-	OwnerRef []metav1.OwnerReference
+	OwnerRef *metav1.OwnerReference
```

Comment:
Why not use the slice here? In most situations there can be more than one OwnerReference.

Reply:
Thats a good question. The object.meta.ownerreferences continue to be a slice. This field refers to the OwnerReference introduced by this controller. The rest of the change is about inserting this controller's reference into the matching objects.